src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -41,4 +41,24 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
+
+    @Override
+    public SizeResponse deleteSize(UUID sizeId){
+        ifSizeExistThrowException(sizeId);
+        Size size = ifSizeIsNotDeletedThrowException(sizeId);
+        size.setDeleted(true);
+
+        return SizeMapper.toSizeResponse(sizeRepository.save(size));
+    }
+
+    public void ifSizeExistThrowException(UUID sizeId){
+        if(!sizeRepository.existsById(sizeId)){
+            throw new NoSuchElementException("Size with id " + sizeId + " does not exist.");
+        }
+    }
```

Comment:
El nombre de este método está mal ya que se lanza la exception si NO existe

```suggestion
    public void ifSizeDoesNotExistsThrowException(UUID sizeId){
        if(!sizeRepository.existsById(sizeId)){
            throw new NoSuchElementException("Size with id " + sizeId + " does not exist.");
        }
    }
```